lib/charliecloud.py

```diff
@@ -456,6 +463,11 @@ def set_(dst_key, *src_keys):
          for k in config["config"]["Volumes"].keys():
             self.metadata["volumes"].append(k)
 
+   def metadata_history_append(self, hist):
+      DEBUG("appending history: %s" % hist)
+      for i in hist:
+         self.metadata['history'].append(i)
```

Comment:
Couple suggestions here:

1. Call it `history_append`?
2. Why does it take a list of items? It's only called once with a singleton.
3. I'd suggest more structured arguments, rather than arbitrary objects appended to `self.metadata["history"]`. E.g., the call I see could just pass the string that ends up in `created_by`.

Reply:
> Call it history_append?

I don't really like the huge name `metadata_history_append` either, however, it allows us to group it alphabetically in charliecloud.py with other related metadata functions.

> Why does it take a list of items? It's only called once with a singleton.

Good catch. I'll update it to take a single dictionary.